### PR TITLE
Build up manpage information on sconscript files.

### PR DESCRIPTION
**Pull request by mwichmann**
Tweaked introductory section with a few more details.

Added a new subsection to begin the _Configuration File Reference_ section, called _Configuration Files_, mainly as a better home for the information on alternate SConstruct names, which seemed a distraction at the top of the file, as well as to introduce the concept that sconscript files are independent of each other, which does not appear to be explicitly stated (although you
could gather the information by reading some Function entries).

Also provides quick links to other Config File subsections - manpage is hard to navigate as it's so long.

This is a doc-only change.

Signed-off-by: Mats Wichmann <mats@linux.com>

## Contributor Checklist:

* [ ] I have created a new test or updated the unit tests to cover the new/changed functionality.
* [ ] I have updated `CHANGES.txt` (and read the `README.rst`)
* [X] I have updated the appropriate documentation

### CODE REVIEW

File: doc/man/scons.xml [hunk 1]
```diff
@@ -102,35 +102,49 @@ orchestrates the construction of software
 (and other tangible products such as documentation files)
 by determining which
 component pieces must be built or rebuilt and invoking the necessary
-commands to build them.</para>
+commands to build them.
+&SCons; offers many features to improve developer productivity
+such as parallel builds, caching of build artefacts, and
```
Comment (bdbaddog) [inline doc/man/scons.xml]: spelling? artifacts

File: doc/man/scons.xml [hunk 2]
```diff
@@ -102,35 +102,49 @@ orchestrates the construction of software
...
+a database of information about previous builds so details do
+not have to be recalculated each run.
+</para>
 
-<para>You instruct
-&scons; by writing a configuration file
-which specifies the files to be built (<firstterm>targets</firstterm>),
-and, if necessary, the rules to build those files.  Premade
-rules exist for building many common software components
+<para>You set up an &SCons;
```
Comment (bdbaddog) [inline doc/man/scons.xml:112]: a SCons ?

File: doc/man/scons.xml [hunk 3]
```diff
@@ -102,35 +102,49 @@ orchestrates the construction of software
...
-configuration from that file.
-An alternate file name may be
-specified via the
-<option>-f</option>
-option.
+in the current directory and reads the
+build configuration from that file
+(other names are possible,
+see <xref linkend="configuration_files"/> for more information).
+An &SConstruct; file is required:
```
Comment (bdbaddog) [inline doc/man/scons.xml]: A SConstruct ?

File: doc/man/scons.xml [hunk 4]
```diff
@@ -2156,8 +2177,57 @@ repositories are searched in the order specified.</para>
 
 <refsect1 id='configuration_file_reference'>
 <title>CONFIGURATION FILE REFERENCE</title>
-<!--  .SS Python Basics -->
-<!--  XXX Adding this in the future would be a help. -->
+
+<refsect2 id='configuration_files'>
+<title>Configuration Files</title>
```
Comment (bdbaddog) [inline doc/man/scons.xml]: Configuration files is likely going to be confusing.  I'd propose SConstruct/SConcript files instead.

File: doc/man/scons.xml [hunk 5]
```diff
@@ -2156,8 +2177,57 @@ repositories are searched in the order specified.</para>
...
+<para>
+The build configuration consists of one or more configuration files
+(SConscript files).
+The main file is named <filename>SConstruct</filename> by
+default, though if necessary, &scons; also checks for alternate spellings
```
Comment (bdbaddog) [inline doc/man/scons.xml]: alternative filenames instead of spellings.

File: doc/man/scons.xml [hunk 6]
```diff
@@ -2156,8 +2177,57 @@ repositories are searched in the order specified.</para>
...
+&Sconstruct;, &sconstruct;, &SConstruct.py;, &Sconstruct.py;
+and &sconstruct.py;
+in that order.
+An alternate file name (including a path to another
```
Comment (bdbaddog) [inline doc/man/scons.xml]: Instead of "An alternate file name" how about "Other file names"..

File: doc/man/scons.xml [hunk 7]
```diff
@@ -3351,12 +3421,11 @@ from SCons.Script import *
 <para>A &consenv; has an associated dictionary of
 <firstterm>&consvars;</firstterm>
 that are used by built-in or user-supplied build rules.
-&Consvar; naming must follow the same rules as for
+&Consvar; naming must follow the same rules as used for
```
Comment (bdbaddog) [inline doc/man/scons.xml]: same rules as Python..

**Comment by bdbaddog** [comment]
Configuration files doesn't sound like an accurate term to me.
To my ears (and of course this is subjective).. configure files always seem like something static and not something which is active and can run logical and iterative operations.

That said I can't think of a better term at the moment.

File: doc/man/scons.xml (hunk 1 above)
Comment (mwichmann) [reply doc/man/scons.xml]: Brtish vs American, but more pointedly, artefact is being used more frequently for the past product of a technical process, as opposed to a piece of physical history.  However, they're virtually interchangeable and I have no problem changing to "artifact"

File: doc/man/scons.xml (hunk 2 above)
Comment (mwichmann) [reply doc/man/scons.xml:112]: SCons starts with a "vowel sound" (ess-cons), so "an" is correct.

File: doc/man/scons.xml (hunk 3 above)
Comment (mwichmann) [reply doc/man/scons.xml]: as previous - unless you want to prefer the consonant-sound pronuciation - I'll have to make a few other changes if so.

File: doc/man/scons.xml (hunk 4 above)
Comment (mwichmann) [reply doc/man/scons.xml]: I can do that, struggling with right term.  Since the containing section was called "Configuration File Reference" I just picked Configuration Files. Happy to change both.

File: doc/man/scons.xml (hunk 5 above)
Comment (mwichmann) [reply doc/man/scons.xml]: I'll make this change.

File: doc/man/scons.xml (hunk 6 above)
Comment (mwichmann) [reply doc/man/scons.xml]: Ok.  A subtle point of `-f` is it changes the algorithm from "look in current directory" to "look in the place I said" if there's a path component, always looking for a better wording to include that without getting too wordy.  Maybe "A different sconstruct file can be specified using the `-f` option"?

File: doc/man/scons.xml (hunk 7 above)
Comment (mwichmann) [reply doc/man/scons.xml]: ok.

**Comment by mwichmann** [comment]
> Configuration files doesn't sound like an accurate term to me.
> To my ears (and of course this is subjective).. configure files always seem like something static and not something which is active and can run logical and iterative operations.
> 
> That said I can't think of a better term at the moment.

Yes.  There's no really good answer, seems every build tool uses slightly different wording here.

File: doc/man/scons.xml (hunk 7 above)
Comment (mwichmann) [reply doc/man/scons.xml]: Actually, come to think of it, since they're only strings used as dictionary keys, is this actually even true?

File: doc/man/scons.xml (hunk 1 above)
Comment (bdbaddog) [reply doc/man/scons.xml]: Please change. I've never seen the artefact used in docs before.. I'm sure someone will file a GH issue about spelling without following instructions  and you know how much I LOVE that... ;)

Comment (mwichmann) [reply doc/man/scons.xml]: done.
